pkg/parquetquery/iters.go

```diff
@@ -119,7 +115,11 @@ func TruncateRowNumber(definitionLevelToKeep int, t RowNumber) RowNumber {
 	case 7:
 		return RowNumber{t[0], t[1], t[2], t[3], t[4], t[5], t[6], t[7]}
 	}
-	level.Error(log.Logger).Log("msg", "definition level out of bound: should be [0:7] but got %d", definitionLevelToKeep)
+
+	if definitionLevelToKeep > MaxDefinitionLevel || definitionLevelToKeep < 0 {
+		level.Error(log.Logger).Log("msg", "definition level out of bound: should be [0:7] but got %d", definitionLevelToKeep)
+		panic("invalid definition level")
+	}
 	return EmptyRowNumber()
```

Comment:
This can just be the default case. This makes it also more clear that the `return EmptyRowNumber()` can never be reached (and thus be removed)